skbuild/constants.py

```diff
@@ -20,10 +20,15 @@ def _default_skbuild_plat_name():
     On macOS, it corresponds to the version and machine associated with :func:`platform.mac_ver()`.
     """
     if sys.platform == 'darwin':
-        # distutils.util.get_platform() returns the release based on the value
-        # of MACOSX_DEPLOYMENT_TARGET on which Python was built, which may
-        # be significantly older than the user's current machine.
+        # If the MACOSX_DEPLOYMENT_TARGET environment variable is defined, use
+        # it, as it will be the most accurate. Otherwise use the value returned
+        # by distutils. distutils.util.get_platform() returns the release based
+        # on the value of MACOSX_DEPLOYMENT_TARGET on which Python was built,
+        # which may be significantly older than the user's current machine.
         release, _, machine = platform.mac_ver()
+        mdt = os.getenv("MACOSX_DEPLOYMENT_TARGET")
+        if mdt:
+            release = mdt
```

Comment:
You could even do this ... 

```python
release = os.environ.get("MACOSX_DEPLOYMENT_TARGET", release)
```

Reply:
Even better. Fixed in 698597d4712350887faef7b1c35d0425f8d88019.